packages/common/src/utils/arrays/utils.ts

```diff
@@ -18,3 +18,20 @@ export const arrayToRecord = <T extends { id: string }>(
     return acc;
   }, {} as Record<string, T>);
 };
+
+// Converts some array of entities to a map of entities
+// where the value of each key is an array of entities
+export const groupBy = <T, K extends keyof T & string>(
+  things: T[],
+  key: K
+): Record<string, T[]> => {
+  const lookup: Record<string, T[]> = {} as Record<string, T[]>;
+  things.forEach(thing => {
+    const value = String(thing[key]);
+    if (!lookup[value]) {
+      lookup[value] = [];
+    }
+    (lookup[value] ?? []).push(thing);
+  });
+  return lookup;
+};
```

Comment:
shifted here from elsewhere which sparked the `lodash` question - will make an issue for adding that package